ansible/tasks/create-configmap.yml

```diff
@@ -19,8 +19,16 @@
         namespace: "{{ test_harness_namespace }}"
         state: present
         apply: true
+      when: not test_harness_dry_run
 
   always:
+    - name: copy configmap file to {{ __test_harness_cm_name }}.yml
+      copy:
+        src: "{{ config_tempfile.path }}"
+        dest: "{{ __test_harness_cm_name }}.yml"
+      when: test_harness_dry_run | bool
+      changed_when: false
+
     - name: Remove config file
       file:
         state: absent
```

Comment:
The secrets file handling looks a little bit strange, now. Where does the tempfile come from and why is it useful? Should the non-temp file be cleaned up in a non-dry run?

Reply:
> The secrets file handling looks a little bit strange, now.

Do you mean "configmap file"?  I'm assuming so in my following answers.

> Where does the tempfile come from

I'm not sure I understand the question.   I don't think you mean "How does the tempfile module work?"  since that's answered in the Ansible documentation.  I'm going to answer the question "Where do the contents of the tempfile come from?"

The contents are generated by the https://github.com/linux-system-roles/test-harness/blob/0c4f6fcae1183e61cf93410b9aa0f9f3ecd556d8/ansible/tasks/create-configmap.yml#L10 using the `oc create configmap --dry-run` command with the config file we want to use in the configmap.

> and why is it useful?

It is useful so that we can feed this file into the `k8s` module which will apply the changes idempotently.  It is also useful so that we can get a copy of the file for debugging and development (e.g. we want to try out some different config options in the staging environment).

> Should the non-temp file be cleaned up in a non-dry run?

I don't think it is a high priority to do this in the Ansible playbook when the user can just rm the file after use.

If someone else feels differently, please submit a followup PR.

Reply:
Let me rephrase, why is a temporary file more useful than always using `"{{ __test_harness_cm_name }}.yml"` instead of copying from the temporary file to that path only in the dry-run case.

Reply:
> Let me rephrase, why is a temporary file more useful than always using `"{{ __test_harness_cm_name }}.yml"` instead of copying from the temporary file to that path only in the dry-run case.

So - don't use a tempfile - always write to `"{{ __test_harness_cm_name }}.yml"` - if not a dry-run, remove the file.

Sure, that would work.